Grid2TrailOfLight.lua

```diff
@@ -17,27 +18,47 @@ local UnitGUID = UnitGUID
 local FlashHealSpellID = 2061
 local TrailOfLightSpellID = 200128 -- it's the talent for the description
 local ToLHealSpellID = 234946 -- it's the spell id for the heal
+local ToLNodeID = 82634 -- talent node id
 
-local TrailOfLightName = GetSpellInfo(TrailOfLightSpellID)
-local TrailOfLightIcon = GetSpellTexture(TrailOfLightSpellID)
+local TrailOfLightName = GetSpellInfo(TrailOfLightSpellID).name
+local TrailOfLightIcon,_ = C_Spell.GetSpellTexture(TrailOfLightSpellID)
 
 --
 local playerGUID = nil
 local CurrentTOLPlayer = nil
 local TrailOfLightSelected = false
-local IsHolySpec = false
 local HealData = {}
 
 TrailOfLight.UpdateAllUnits = Grid2.statusLibrary.UpdateAllUnits
 
+-- Utility to check if ToL is talented
+local function IsToLTalented()
+    local configID = C_ClassTalents.GetActiveConfigID()
+    if configID == nil then return end
+
+    local configInfo = C_Traits.GetConfigInfo(configID)
+    if configInfo == nil then return end
+	
+	-- Check holy is current spec
+	if configInfo.name ~= "Holy" then return end	
```

Comment:
The check will only work on english clients. For example, on my french client `configInfo.name` will return "Sacré" instead of "Holy". 
We can use ``if GetSpecialization() ~= 2 then return end`` for avoiding issues with the localization.